packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1120,4 +1138,50 @@ apiDescribe('Database', (persistence: boolean) => {
       }).to.throw();
     });
   });
+
+  it('can wait for pending writes as expected', async () => {
+    await withTestDoc(persistence, async docRef => {
+      const firestore = docRef.firestore;
+      // Prevent pending writes receiving acknowledgement.
+      await firestore.disableNetwork();
+
+      const awaitPendingWrites1 = waitForPendingWrites(firestore);
+      const pendingWrites = docRef.set({ foo: 'bar' });
+      const awaitPendingWrites2 = waitForPendingWrites(firestore);
+
+      // `awaitsPendingWrites1` resolves immediately because there are no pending writes at
+      // the time it is created.
+      await expect(awaitPendingWrites1).to.be.eventually.fulfilled;
```

Comment:
Hm. I don't know how you can check in JS if a Promise is already resolved. If you want to keep this line (I think you should) I would adjust the comment to explain why the Promise does resolve even though the network is offline.